numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1464,8 +1430,121 @@ array_richcompare(PyArrayObject *self, PyObject *other, int cmp_op)
                 n_ops.greater_equal);
         break;
     default:
-        result = Py_NotImplemented;
-        Py_INCREF(result);
+        Py_INCREF(Py_NotImplemented);
+        return Py_NotImplemented;
+    }
+    if (result == NULL) {
+        /*
```

Comment:
Can you extract the contents of this long `if` to `_failed_compare_workaround(self, other, op)`, or something similar?

Reply:
Done.